src/lib/components/event/metadata-decoder.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+
+  import type { Payload } from '$lib/types';
+  import { decodeSingleReadablePayloadWithCodec } from '$lib/utilities/decode-payload';
+  import {
+    getCodecEndpoint,
+    getCodecIncludeCredentials,
+    getCodecPassAccessToken,
+  } from '$lib/utilities/get-codec';
+
+  export let value: Payload | undefined = undefined;
+  export let fallback: string = '';
+  export let onDecode: (decodedValue: string) => void | undefined = undefined;
+
+  let decodedValue = fallback;
+
+  $: endpoint = getCodecEndpoint($page.data.settings);
+  $: passAccessToken = getCodecPassAccessToken($page.data.settings);
+  $: includeCredentials = getCodecIncludeCredentials($page.data.settings);
+  $: settings = {
+    ...$page.data.settings,
+    codec: {
+      ...$page.data.settings?.codec,
+      endpoint,
+      passAccessToken,
+      includeCredentials,
+    },
+  };
+
+  const decodePayload = async (_value: Payload | undefined) => {
+    if (!_value) {
+      return;
+    }
+
+    try {
+      const metadata = await decodeSingleReadablePayloadWithCodec(
+        _value,
+        settings,
+      );
+
+      if (typeof metadata === 'string') {
+        decodedValue = metadata;
+      }
+
+      if (onDecode) {
+        onDecode(decodedValue);
+      }
+    } catch (e) {
+      console.error('Could not decode payloads');
+    }
+  };
+
+  $: decodePayload(value);
+</script>
+
+<slot {decodedValue} />
```

Comment:
I think it makes more sense to have `decodePayload` return the decoded value instead of mutating local state. Then this would be

```
{#await decodeValue(value) then decodedValue}
  <slot {decodedValue} />
{/await}
```

Could then also add a slot for an error or something if desired.

Reply:
Good call, for now I'm going to use the fallback for the :catch block value